### PR TITLE
[Merged by Bors] - forward port changes to Algebra.Order.Hom.Basic

### PR DESCRIPTION
**Pull request by j-loreaux**
This forward ports changes to this file from [mathlib3#16919](https://github.com/leanprover-community/mathlib/pull/16919). This does not address the other files in that PR, but #1238 is open to address those. This is being PR'ed separately so that work on `Analysis.Normed.Group.Seminorm` can continue (#2400).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Hom/Basic.lean [hunk 1]
```diff
@@ -122,3 +166,201 @@ theorem le_map_div_add_map_div [Group α] [AddCommSemigroup β] [LE β] [MulLEAd
...
+
+export AddGroupNormClass (eq_zero_of_map_eq_zero)
+
+export GroupNormClass (eq_one_of_map_eq_zero)
+
+attribute [simp] map_one_eq_zero -- porting note: `to_additive` translation already exists
+
+attribute [simp] map_neg_eq_map
+
+attribute [simp] map_inv_eq_map -- porting note: `to_additive` translation already exists
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Hom/Basic.lean:229]: @YaelDillies, do you know what's going on here? Were the attributes in error in Lean 3?

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Hom/Basic.lean:229]: `to_additive` in mathlib3 doesn't throw an error when the declaration already exists. That might be it?

Comment (jcommelin) [reply Mathlib/Algebra/Order/Hom/Basic.lean:229]: So this was wrong in ml3?

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Hom/Basic.lean:229]: No, it's expected behavior. I think it's just that the syntax changed to `to_additive existing the_existing_lemma`.

**Comment by ChrisHughes24** [comment]
bors merge

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:229]: Does that mean we need to add `to_additive existing` in this PR?

**Pull request closed without merging**